txscript/engine.go

```diff
@@ -335,8 +335,6 @@ func (vm *Engine) isBranchExecuting() bool {
 // conditional).
 func isOpcodeDisabled(opcode byte) bool {
 	switch opcode {
-	case OP_CAT:
```

Comment:
Similar to the [discussion in Inquistion](https://github.com/bitcoin-inquisition/bitcoin/pull/39#discussion_r1464005864), this check must still be triggered for legacy script, since the opcode won't be executed if inside a conditional.

Reply:
Done: https://github.com/btcsuite/btcd/pull/2095/commits/30ed9eccc127faccbb47f3e03a92825366c9d71f